src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java

```diff
@@ -71,12 +71,15 @@ public DataFetcherResult getRecords(int maxRecords) {
         
         if (nextIterator != null) {
             try {
-                return new AdvancingResult(kinesisProxy.get(nextIterator, maxRecords));
+                GetRecordsResult getRecordsResult = kinesisProxy.get(nextIterator, maxRecords);
+                LOG.info("Successfully fetched records from Kinesis for shard " + shardId + ": " + getRecordsResult);
```

Comment:
Logging every response at INFO level can be too much for existing applications. Let's log at INFO only when nextIterator is null, which is the interesting case.

Reply:
Made the change